src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+                    break;
+
+                case EVENT_WORMCRUD_CONVERSATION_COOLDOWN:
+                    Reset();
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    void CheckForSeaOrcs()
+    {
+        me->GetCreatureListWithEntryInGrid(_orcSeaDogList, NPC_DRUNKEN_ORC_SEA_DOG, 5.0f);
+
+        for (uint8 i = 0; i < 3; i++)
+        {
+            for (Creature* seaDogOrc : _orcSeaDogList)
+                _drunkenOrcSeaDog[i] = seaDogOrc->GetGUID();
+        }
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+    std::array<ObjectGuid, 3> _drunkenOrcSeaDog;
+    std::vector<Creature*> _orcSeaDogList;
+    bool _eventCD;
+};
+
+enum HatchetRearGuard
+{
+    QUEST_STEEL_THUNDER                         = 27069,
+    QUEST_LOST_IN_THE_DARKNESS                  = 27093,
+
+    NPC_ORC_SEA_DOG                             = 44942,
+    NPC_WARLORD_TOROK                           = 44917,
+
+    SPELL_SUMMON_ORC_SEA_PUP                    = 83839,
+    SPELL_SEA_PUP_TRIGGER                       = 83865,
+
+    EVENT_HATCHET_CHECK_CONVERSATION            = 1,
+    EVENT_HATCHET_CONVERSATION_COOLDOWN         = 2,
+    EVENT_HATCHET_TALK                          = 3,
+
+    TALK_HATCHET_0                              = 0,
+    TALK_HATCHET_1                              = 1,
+    TALK_TOROK_0                                = 0,
+    TALK_TOROK_1                                = 1,
+    TALK_TOROK_2                                = 2,
+
+    ANIMKIT_TOROK                               = 594
+};
+
+// 44916 - Admiral Hatchet
+struct npc_silverpine_admiral_hatchet : public ScriptedAI
+{
+    npc_silverpine_admiral_hatchet(Creature* creature) : ScriptedAI(creature), _eventCD(false) { }
+
+    void JustAppeared() override
+    {
+        CheckForTorok();
+    }
+
+    bool OnGossipSelect(Player* player, uint32 /*menuId*/, uint32 /*gossipListId*/) override
+    {
+        if (player->GetQuestStatus(QUEST_STEEL_THUNDER) == QUEST_STATUS_INCOMPLETE)
+            player->CastSpell(player, SPELL_SUMMON_ORC_SEA_PUP);
+
+        CloseGossipMenuFor(player);
+        return false;
+    }
+
+    void OnQuestAccept(Player* player, Quest const* quest) override
+    {
+        if (quest->GetQuestId() == QUEST_STEEL_THUNDER)
+            player->CastSpell(player, SPELL_SUMMON_ORC_SEA_PUP);
+    }
+
+    void OnQuestReward(Player* player, Quest const* quest, LootItemType /*type*/, uint32 /*opt*/) override
+    {
+        if (quest->GetQuestId() == QUEST_STEEL_THUNDER)
+            player->CastSpell(player, SPELL_SEA_PUP_TRIGGER, true);
+
+        if (quest->GetQuestId() == QUEST_LOST_IN_THE_DARKNESS)
+        {
+            me->GetCreatureListWithEntryInGrid(_seaDogList, NPC_ORC_SEA_DOG, 25.0f);
+
+            for (Creature* seadog : _seaDogList)
+            {
+                if (seadog->GetOwner()->ToPlayer()->IsQuestRewarded(QUEST_LOST_IN_THE_DARKNESS))
```

Comment:
what if owner is null? (e.g. gm spawning such a creature)
what if owner is no player?

Reply:
Feedback applied. There was a Despawn All Summons spells being cast that I missed.